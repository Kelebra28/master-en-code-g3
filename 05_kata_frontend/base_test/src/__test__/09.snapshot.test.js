import { getCharacter } from '../snapshot';
import rick from '../rick.json';

describe('Voy a realizar un snapshot', () => {
  test('Snapshot de mi objeto json', () => {
    const character = getCharacter(rick);
    
    expect(character).toMatchSnapshot();
  })

  // test('Siempre falla la captura', () => {
  //   const user  = {
  //     createAt: new Date(),
  //     id: Math.floor(Math.random() * 20)
  //   }

  //   expect(user).toMatchSnapshot();
  // })

  test('Realizar excepciones dentro de la captura', () => {
    const user  = {
      name: "Mikee Ramirez",
      id: Math.floor(Math.random() * 20)
    };

    expect(user).toMatchSnapshot({
      id: expect.any(Number),
    })
  })
})