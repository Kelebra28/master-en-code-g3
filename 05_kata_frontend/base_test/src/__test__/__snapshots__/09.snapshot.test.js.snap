// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Voy a realizar un snapshot Realizar excepciones dentro de la captura 1`] = `
Object {
  "id": Any<Number>,
  "name": "Mikee Ramirez",
}
`;

exports[`Voy a realizar un snapshot Snapshot de mi objeto json 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rick Sanchez",
  "species": "Human",
  "status": "Alive",
}
`;
